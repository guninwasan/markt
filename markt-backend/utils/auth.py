from functools import wraps
from flask import request, jsonify
from database.models import ApiKey  # Import your ApiKey model

# API Key validation helper function
def validate_api_key(api_key, user_id):
    api_key_record = ApiKey.query.filter_by(key=api_key, user_id=user_id).first()
    return api_key_record is not None  # Return True if a matching record is found

# Decorator to protect routes with API key and user ID
def api_key_required(f):
    @wraps(f)
    def decorated_function(*args, **kwargs):
        api_key = request.headers.get('API-Key')      # Extract API key from headers
        user_id = request.headers.get('User-ID')      # Extract User-ID from headers
        
        # Check if both headers are present and valid
        if not api_key or not user_id or not validate_api_key(api_key, user_id):
            return jsonify({"status": "error", "message": "Unauthorized: Invalid API Key or User ID"}), 401
        
        return f(*args, **kwargs)
    
    return decorated_function
