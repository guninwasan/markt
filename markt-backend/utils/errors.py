from enum import Enum, auto
# NOTE: after updating this file, run python syncErrors.py to sync with errorCodes.ts

class ErrorRsp(Enum):
    OK = 1000
    ERR_PARAM = auto()
    ERR_PARAM_DUP = auto()
    ERR_PARAM_EMAIL = auto()
    ERR_PARAM_PWD = auto()
    ERR_PARAM_PHONE = auto()
    ERR_NOT_FOUND = auto()
    ERR = auto()
