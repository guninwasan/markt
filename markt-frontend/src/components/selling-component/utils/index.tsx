export * from "./upload-image";
export * from "./valid-form-details";
