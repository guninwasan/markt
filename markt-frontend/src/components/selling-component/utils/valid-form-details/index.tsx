export * from "./valid-form-details";
