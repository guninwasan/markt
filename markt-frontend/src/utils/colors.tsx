const colors = {
  primary: "#b8d3e0",
  textBlack: "#333",
  black: "#333",
  lightGrey: "#d9d9d9",
  white: "#f9f9f9",
};

export { colors };
