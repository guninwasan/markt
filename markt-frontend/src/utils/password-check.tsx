const passwordCheck = (password: string) => {
  const isAtLeast8Chars = password.length >= 8;
  const hasLowerCase = /[a-z]/.test(password);
  const hasUpperCase = /[A-Z]/.test(password);
  const hasNumber = /\d/.test(password);
  const hasSpecialChar = /[@$!%*?&#-]/.test(password);

  return (
    isAtLeast8Chars &&
    hasLowerCase &&
    hasUpperCase &&
    hasNumber &&
    hasSpecialChar
  );
};

export { passwordCheck };
