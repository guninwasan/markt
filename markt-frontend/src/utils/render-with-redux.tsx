import { render } from "@testing-library/react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux";

export const renderWithRedux = (component: React.ReactNode) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>{component}</MemoryRouter>
    </Provider>
  );
};
