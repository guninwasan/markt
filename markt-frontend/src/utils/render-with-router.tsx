import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

export const renderWithRouter = (component: React.ReactNode) => {
  return render(<BrowserRouter>{component}</BrowserRouter>);
};
