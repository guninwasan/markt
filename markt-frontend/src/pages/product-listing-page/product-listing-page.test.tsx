import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { ProductListingPage } from ".";
import { renderWithRouter } from "../../utils/render-with-router";

jest.mock("../../components", () => ({
  Navbar: () => <nav data-testid="navbar">nav</nav>,
  ProductListingComponent: () => (
    <div data-testid="product-listing-component" />
  ),
}));

describe("ProductListingPage", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("renders without crashing", () => {
    renderWithRouter(<ProductListingPage />);
  });

  it("renders Navbar component", () => {
    renderWithRouter(<ProductListingPage />);

    // Check for the presence of the navigation role
    const navbar = screen.getByText("nav");
    expect(navbar).toBeInTheDocument();
  });

  it("renders ProductListingComponent component", () => {
    renderWithRouter(<ProductListingPage />);

    expect(screen.getByTestId("product-listing-component")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = renderWithRouter(<ProductListingPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the correct number of children in the main div", () => {
    renderWithRouter(<ProductListingPage />);
    const mainDiv = screen.getByRole("navigation", {
      hidden: true,
    }).parentElement;
    expect(mainDiv?.children.length).toBe(2);
  });
});
