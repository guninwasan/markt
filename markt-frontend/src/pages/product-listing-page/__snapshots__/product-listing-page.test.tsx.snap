// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListingPage matches snapshot 1`] = `
<div>
  <nav
    data-testid="navbar"
  >
    nav
  </nav>
  <div
    data-testid="product-listing-component"
  />
</div>
`;
